### DISCUSSION TITLE
emane integration issues

### DISCUSSION THREAD

**Discussion by PatrickTague** [coreemu/core#629] [Q&A]
maybe this is an easy thing to figure out, but i'm not able to find anything, so checking to see if others have had this same issue.

when using WLAN devices with EMANE in the CORE GUI, then starting a session, then right-clicking on a WLAN-connected device and selecting to start Wireshark on the corresponding interface, Wireshark launches but then gives an error that the corresponding veth interface doesn't exist.  is there a good way to identify which virtual interface that Wireshark sees corresponds to the virtual EMANE WLAN interface?

thanks in advance!

----------

[edit] in continuing to use EMANE, i'm seeing rather a lot of EMANE features that aren't working, despite being installed using the installation method given in the CORE documentation.  i'm getting error messages in trying to use things like `emanesh` and others, and digging in the error report details, it says that emane may not be installed correctly.  is there a deeper issue with the integration, or is there a simple fix to get the EMANE features to work?

**Comment by bharnden** [comment]
Sorry for delay in getting back here. 

First the old GUI and some of the integrations may be flawed, I would run wireshark directly within a node if desired, or on the actual host.

Second, there are some nuances that the current CORE provided installation does not account for, for some of the EMANE tooling. CORE by default goes into a python virtual environment, the install helps to account for this and installs EMANE, then install the python binding into the virtual environment. Some of EMANEs tools depend on the same python bindings, but they expect them to be on the host system and they are not present, which is likely the issue you run into.

I will likely be removing the general convenience of installing EMANE and refer to installing EMANE through their pre-built packages: https://github.com/adjacentlink/emane/wiki/Install

I may keep the convenience install for installing the EMANE python bindings into the virtual environment alone, and nothing else.

Alternatively, if you installed CORE locally using `-l` and the CORE provided EMANE install using `-l` that should resolve that part.
